src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -464,64 +306,82 @@ private TestExecutionResult testPluginAgainst(
-                        LOGGER.log(
-                                Level.INFO,
-                                "Copy plugin directory from {0}",
-                                config.getLocalCheckoutDir().getAbsolutePath());
+                        cloneFromScm(
+                                pomData.getConnectionUrl(),
+                                config.getFallbackGitHubOrganization(),
+                                getScmTag(pomData, plugin.name, plugin.version),
+                                pluginCheckoutDir);
+                    }
+                } else {
+                    // TODO this fails when it encounters symlinks (e.g.
+                    // work/jobs/…/builds/lastUnstableBuild), and even up-to-date versions of
+                    // org.apache.commons.io.FileUtils seem to not handle links, so may need to
+                    // use something like
+                    // http://docs.oracle.com/javase/tutorial/displayCode.html?code=http://docs.oracle.com/javase/tutorial/essential/io/examples/Copy.java
+                    LOGGER.log(
+                            Level.INFO,
+                            "Copy plugin directory from {0}",
+                            config.getLocalCheckoutDir().getAbsolutePath());
+                    try {
                         org.codehaus.plexus.util.FileUtils.copyDirectoryStructure(
                                 config.getLocalCheckoutDir(), pluginCheckoutDir);
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
                     }
-                } else {
-                    // These hooks could redirect the SCM, skip checkout (if multiple plugins use
-                    // the same preloaded repo)
-                    cloneFromSCM(pomData, plugin.name, plugin.version, pluginCheckoutDir, "");
                 }
             } else {
-                // If the plugin exists in a different directory (multi-module plugins)
-                if (beforeCheckout.get("pluginDir") != null) {
-                    pluginCheckoutDir = (File) beforeCheckout.get("checkoutDir");
-                }
-                if (beforeCheckout.get("parentFolder") != null) {
-                    parentFolder = (String) beforeCheckout.get("parentFolder");
-                }
-                LOGGER.log(
-                        Level.INFO,
-                        "The plugin has already been checked out, likely due to a multi-module"
-                                + " situation; continuing");
+                // These hooks could redirect the SCM, skip checkout (if multiple plugins use
+                // the same preloaded repo)
+                cloneFromScm(
+                        pomData.getConnectionUrl(),
+                        config.getFallbackGitHubOrganization(),
+                        getScmTag(pomData, plugin.name, plugin.version),
+                        pluginCheckoutDir);
```

Comment:
While I was here I deduplicated some logic that was duplicated in the various callers of `cloneFromSCM`.